aten/src/ATen/native/Fill.cpp

```diff
@@ -18,7 +18,7 @@ Tensor& fill_out(Tensor& self, Scalar value) {
     self.copy_(out);
     return self;
   }
-  if (self.device() == at::kCPU && self.numel() == 1 && !self.is_complex() && !value.isComplex()) {
+  if (self.device().type() == at::kCPU && self.numel() == 1 && !self.is_complex() && !value.isComplex()) {
```

Comment:
Seems unnecessary to cast to a `Device` for the comparison instead of using the `DeviceType` (although we'd get the perf improvement regardless).